### PR TITLE
feat(minor-multisig-prover)!: restrict update_workerset to admin only

### PR DESCRIPTION
**Pull request by cjcobb23**
https://axelarnetwork.atlassian.net/browse/AXE-3321

### CODE REVIEW

File: contracts/multisig-prover/src/execute.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 use cosmwasm_std::{
-    to_binary, wasm_execute, Addr, DepsMut, Env, QuerierWrapper, QueryRequest, Response, Storage,
-    SubMsg, WasmQuery,
+    to_binary, wasm_execute, Addr, DepsMut, Env, MessageInfo, QuerierWrapper, QueryRequest,
+    Response, Storage, SubMsg, WasmQuery,
```
Comment (cgorenflo) [inline contracts/multisig-prover/src/execute.rs:3]: why are all of those needed?

**Review by cgorenflo — APPROVED** [review]

**Comment by cgorenflo** [comment]
remember to give the commit an appropriate `minor` tag

**Review by fish-sammy — COMMENTED** [review]
The PR is a `feat` instead of `refactor`.

File: contracts/multisig-prover/src/execute.rs [hunk 2]
```diff
@@ -17,6 +17,14 @@ use crate::{
     types::{BatchId, WorkersInfo},
 };
 
+pub fn require_admin(deps: &DepsMut, info: MessageInfo) -> Result<(), ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    if config.admin != info.sender {
+        return Err(ContractError::Unauthorized);
+    }
+    Ok(())
```
Comment (fish-sammy) [inline contracts/multisig-prover/src/execute.rs:25]:
```suggestion
    match CONFIG.load(deps.storage)?.admin {
        admin if admin == info.sender => Ok(()),
        _ => Err(ContractError::Unauthorized),
    }
```

**Review by fish-sammy — APPROVED** [review]

File: contracts/multisig-prover/src/execute.rs (hunk 1 above)
Comment (cjcobb23) [reply contracts/multisig-prover/src/execute.rs:3]: It's just `MessageInfo` that's added actually

**Pull request merged**